src/libraries/Directory.Build.targets

```diff
@@ -299,4 +299,49 @@
     </PropertyGroup>
   </Target>
 
+  <PropertyGroup>
+    <BeforePack>IncludeAnalyzersInPackage;$(BeforePack)</BeforePack>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <!-- Ensure AnalyzerReference items are restored and built 
+         The target framework of Analyzers has no relationship to that of the refrencing project, 
+         so we don't apply TargetFramework filters nor do we pass in TargetFramework.
+         When BuildProjectReferences=false we make sure to set BuildReference=false to make
+         sure not to try to call GetTargetPath in the outerbuild of the analyzer project. -->
+    <ProjectReference Include="@(AnalyzerReference)" SkipGetTargetFrameworkProperties="true" UndefineProperties="TargetFramework" ReferenceOutputAssembly="false" PrivateAssets="all" BuildReference="$(BuildProjectReferences)"  />
```

Comment:
Why is this called `AnalyzerReference`? Presumably a source generator isn't always an analyzer. Also subfolder that we standardized on in the libs tree is called `gen` an not `analyzer`.

Reply:
Analyzers and source generators are both communicated as "analyzer" assemblies.  https://github.com/dotnet/roslyn/blob/main/docs/compilers/CSharp/CommandLine.md

The NuGet conventions they use, MSBuild items, and task parameters all refer to them as analyzers.  I think it makes more sense to be consistent rather than imply there is something different with a generator.  If roslyn wanted to invent a new term here that might be better (like compiler extension or something).